### ISSUE TITLE
extract_animation.py exports wrong framerange

### ISSUE THREAD

**Issue by LegacyID1991** [mindbender-studio/core#93]
### Problem:
Seams that the _extract_animation.py_ doesn't export the correct frame range

This bit of code doesn't seam to generate the correct frame range.
```
frame_range=(instance.data["startFrame"],
             instance.data["endFrame"])
```

### Solution

The code could be changed to this:
```
frame_range=(int(cmds.playbackOptions(query=True, animationStartTime=True)),
             int(cmds.playbackOptions(query=True, animationEndTime=True)))
```
Because that info gets checked by another validator before this code executes.

Maybe there is a better or Cooler way of doing it.
What do you think ?

**Comment by mottosso** [comment]
The time is currently coming from the Instance in Maya. In the attribute editor, Extra Attributes at the bottom, you've got `Start Frame` and `End Frame`.

Those values are coming from [here](https://github.com/mindbender-studio/core/blob/master/mindbender/maya/pipeline.py#L176) which should be doing what you just proposed, but isn't.

Hasn't this been a problem before? This looks suspicious.

As a temporary workaround, whatever the artist sets those values to will be the ones exported in the Alembic.

**Comment by LegacyID1991** [comment]
The data of `Start Frame` and `End Frame` is currently:
`Start Frame = 1.0`
`End Frame = 200.0`

as for has this been a problem before.
This was brought to my attention today. 
>[14:00:15] calle_meindbender: verkar som att alembicen inte tar ut det som är satt igen
[14:00:19] calle_meindbender: börjar från 0
[14:00:33] divrA: okey.
[14:00:45] divrA: jag kollar
[14:01:47] calle_meindbender: den kär från 0 till 200 typ

**Comment by LegacyID1991** [comment]
I could create another validator that makes sure that the value is set correctly.

It will just be a copy of the _validate_project_editorial_info.py_ except it will work on instances instead of context.

**Comment by mottosso** [comment]
For posterity, those issues are:

1. Frame range for once version of an Alembic differs from the frame range of another.
2. Publishing character from one shot in another shot will cause the wrong frame range to be used.
3. Accidental frame range differences due to what an artist may do

The value is written (and stored) with the character so as to avoid these issues. But again, it may not apply to you and the only way to know for sure is to test.

**Comment by LegacyID1991** [comment]
I am not sure I understand the need to store the frame-range with the alembic.

In case of animation why shouldn't the scene always be setup to have the latest editorial information ? to always produce the exact **IN** and **OUT** of that shot ?

**Comment by mottosso** [comment]
That's ok. I would try it your way, odds are you are right. It's difficult for me to say without being on-site.

**Comment by LegacyID1991** [comment]
### edit on that:
Of cause you should store the frame range with the alembic.

But why should you publish from one shot to another shot ?
if you are working on `shot 010` I expect all that that you produce from that scene is suppose to work for `shot 010` and not any other shot.

**Comment by LegacyID1991** [comment]
Maybe there is a cool middle ground here.

What if the values in the instance data can be used as a override for the frame range.
If their values are set to 0.0 then they wont be used in the extract animation and instead you use the data from context ?

or will that never be used ?

**Comment by LegacyID1991** [comment]
> 1. Frame range for one version of an Alembic differs from the frame range of another.

True but I cant think of anytime that you would wanna keep the old version with its old frame range.
If you really wanna then you can just add a new shot with the new frame range and then copy all of your old scenes and assets into the new shot and publish that separately.

**Comment by CalleHalldin** [comment]
I'd say lets try having the exporter looking at the .Bat for frame length but have it be optional, Only warn if its not matching. there might be situations like Markus says where you might not want to export something for the full length of the shot because its heavy and only visible the last 10% of the shot, in that case it can warn you but lets you proceed you still wants too. lets give i a go like Marcus said.

**Comment by mottosso** [comment]
Fixed with #96.
